core/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -654,7 +685,10 @@ public void run() {
                 logger.error("Processing loop ended due to uncaught exception. Processor pausing.", e);
                 state.set(State.PAUSED_ERROR);
             } finally {
-                activeSegments.remove(segment.getSegmentId());
+                releaseSegment(segment.getSegmentId());
+                if (activeSegments.size() == maxThreadCount - 1) {
+                    new WorkerLauncher().run();
```

Comment:
Shouldn't we use `threadFactory` for spawning this thread?

Reply:
It is already a thread build by `threadFactory`

Reply:
Not sure that I follow, you are instantiating it here. I was referring to something like: `threadFactory.newThread(new WorkerLauncher()).start();`. Am I missing something?

Reply:
Aha, sorry, the thread which is starting the `WorkerLauncher` is built by `threadFactory`. You can disregard this comment :)